### ISSUE TITLE
Configurable gesture/button settings for normal/reader/zen mode

### ISSUE THREAD

**Issue by dstango** [ddvk/remarkable-hacks#103]
As more and more wishes and preferences about special gestures a button actions show up as issues I"m wondering wether it would be feasible to have a configuration screen in which everybody could self-configure their preferred actions for all sorts of button/gestures in the various modes (normal, reader, zen). 
Something like (examples for current defauls)
Button or gesture / Normal / Reader / Zen
Right / page forward / page forward / toggle last used pen
Left / page back / page back / toggle eraser
Left-Long / first page / first page / undo
...

That would give tremendous flexibility to individual preferred workflows. 

Possible extensions:
- make settings permanent accross rm updates
- provide preference sets with typical combinations of settings for easy selection
- save as / load from -- for sets of settings to easily change

Probably it's asking too much, but it would be great :-)

Thanks you for your constant efforts and support! It's already great!

**Comment by fedecupe** [comment]
Just wanted to add a +1: I am on rm2 and would like to access read mode, but from the readme it seems the only access is via rm1 physical buttons.

Thanks for your work.

**Comment by folofjc** [comment]
@fedecupe you can still get into the read mode on the rM2 by clicking on the battery/document icon in the bottom left, and then clicking "M".

**Comment by b4be1** [comment]
@folofjc, thanks for pointing out how to turn on the reading mode on rM2. Do you know if there is a way to use Zen mode on rM2 too? I can turn it on, but since there are no buttons, I cannot toggle writing/erasing and I cannot undo actions.

**Comment by b4be1** [comment]
Ok, I found it https://github.com/ddvk/remarkable-hacks/issues/93#issuecomment-719584931

**Comment by b4be1** [comment]
I think it would make sense to toggle the modes [normal, zen, reading] all by the same gesture, e.g., long press on the toggle icon (actually short press would be even better). Because now if do long press on the toggle icon while in reading mode, I get into normal mode. But I cannot get back into reading mode in the same way, which is somewhat counterintuitive.

**Comment by folofjc** [comment]
@b4be1 FWIW I prefer it not to toggle all three. When I go into reading mode, I will leave it in reading mode while I am reading. However, when in a document, I toggle zen and normal all the time (to switch between pens, to zoom, to select something, etc). However, the ability to change what the buttons do on the rM1 (the original request) seems reasonable.

**Comment by b4be1** [comment]
Yes, it would be ideal to have the ability to configure gestures/buttons.

Regarding toggling normal/zen mode, on rM2 there actually seems to be no need for normal mode. Swiping down in zen mode toggles the side bar, and thus reveals all options which are also available in normal mode. So zen mode is strictly better than normal mode in my experience. Maybe on rM1 the situation is different because buttons have different meaning in normal/zen modes.

**Comment by folofjc** [comment]
@b4be1 This is true, however I tend to not use it because I tend to swipe down too close to the top and it closes my doc, so I just got used to clicking the button. But that is probably just me.
